finality-aleph/src/data_io/mod.rs

```diff
@@ -24,34 +24,22 @@ pub const MAX_DATA_BRANCH_LEN: usize = 7;
 
 /// The data ordered by the Aleph consensus.
 #[derive(Clone, Debug, Encode, Decode)]
-pub enum AlephData<B: BlockT> {
-    Empty,
-    HeadProposal(UnvalidatedAlephProposal<B>),
+pub struct AlephData<B: BlockT> {
+    pub head_proposal: UnvalidatedAlephProposal<B>,
 }
 
 // Need to be implemented manually, as deriving does not work (`BlockT` is not `Hash`).
 impl<B: BlockT> Hash for AlephData<B> {
     fn hash<H: Hasher>(&self, state: &mut H) {
-        match self {
-            AlephData::Empty => {
-                (0u8).hash(state);
-            }
-            AlephData::HeadProposal(proposal) => {
-                (1u8).hash(state);
-                proposal.hash(state);
-            }
-        }
+        (1u8).hash(state);
```

Comment:
this line is not needed anymore

Reply:
You can also try to derive `Hash` -- previously it was not possible, but we are a couple of versions of rustc ahead.

Reply:
Removed the line but unfortunately, BlockT contains sp_std::hash::Hash and not std::hash::Hash and derive still don't work.